keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,302 @@
+            trainable=trainable,
+        )
+        # All references to `self` below this line
+        self.token_embedding = (
+            token_and_position_embedding_layer.token_embedding
+        )
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        # BOS token '<s>' is equivalent to '[CLS]' from BERT
+        self.bos_token_index = bos_token_index
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "dropout": self.dropout,
+                "bos_token_index": self.bos_token_index,
+            }
+        )
+        return config
+
+
+# TODO: add RobertaMLM, different from BERT.
+
+
+class RobertaClassifier(keras.Model):
+    """Roberta encoder model with a classification head.
+
+    Args:
+        base_model: A `keras_nlp.models.Roberta` to encode inputs.
+        num_classes: Int. Number of classes to predict.
+        hidden_dim: Int. The size of the pooler layer.
+        bos_token_index: Int. Index of <s> token in the vocabulary. Equivalent
+            to [CLS] in BERT.
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```
+    python
+    # Randomly initialized Roberta encoder
+    encoder = keras_nlp.models.Roberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call classifier on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=50265),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    classifier = keras_nlp.models.RobertaClassifier(encoder, 4)
+    logits = classifier(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        base_model,
+        num_classes,
+        hidden_dim=None,
+        bos_token_index=0,
+        dropout=0.1,
+        name=None,
+        trainable=True,
+    ):
+        inputs = base_model.input
+        if hidden_dim is None:
+            hidden_dim = base_model.hidden_dim
+
+        x = base_model(inputs)[:, bos_token_index, :]
+        x = keras.layers.Dropout(dropout, name="pooler_dropout_1")(x)
```

Comment:
These names collectively pooler_dropout_1, pooled_dense and pooler_dropout_2 are a bit confusing as a set. Can we make these clearer?